### DISCUSSION TITLE
Problems with running self-constructed datasets

### DISCUSSION THREAD

**Discussion by mssssss123** [mlfoundations/open_clip#646] [Announcements]
Hi, thank you for such a great job！

I used the img2dataset library to convert the image-text dataset I built into the webdataset format, including jpg, txt, and json files. But when I use openclip for training, I not only want it to use jpg and txt files for training, I additionally save another type of text in the json file. I hope to be able to divide it into three situations during training, one is to use text in txt, one is to use text in an attribute in json, and one is to use a mixture of the two.

Can you give me some suggestions on how to use it or modify the code?

**Comment by gabrielilharco** [comment]
Hi @mssssss123. You can modify the webdataset pipeline to do that. In particular in this line https://github.com/mlfoundations/open_clip/blob/f692ec95e1bf30d50aeabe2fd32008cdff53ef5e/src/training/data.py#L391 we map the file extensions to the image and text tower. So you can use different keys for the text argument there depending on what you want to do.

**Comment by EIFY** [comment]
@mssssss123 You may want to take a look at https://github.com/EIFY/open_clip/commit/66e4603fe06be1bb9547d8ffa999858117d47b2b, which enables open_clip to use Redcaps json caption files.
